Gems/OpenXRVk/Code/Source/InputDeviceXRController.cpp

```diff
@@ -419,4 +450,307 @@ namespace AzFramework
+        debugDisplay.SetColor(axisColorX);
+        debugDisplay.DrawLine(controllerPosition, controllerPosition + controllerTransform.GetBasisX());
+        debugDisplay.SetColor(axisColorY);
+        debugDisplay.DrawLine(controllerPosition, controllerPosition + controllerTransform.GetBasisY());
+        debugDisplay.SetColor(axisColorZ);
+        debugDisplay.DrawLine(controllerPosition, controllerPosition + controllerTransform.GetBasisZ());
+    }
+
+#endif // !_RELEASE
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    void InputDeviceXRController::CheckDebugDrawCheat() const
+    {
+        // This looks for specific controller input and will toggle the debug draw cvar.
+        const auto& rawControllerData = m_impl->GetRawState();
+        using xrc = InputDeviceXRController;
+        static bool cheatWasPressed = false;
+
+        // Menu button + Left Trigger pulled past 0.9 will toggle.
+        // To avoid button bounce, block re-toggle until the menu button is released.
+        const bool menuPressed = rawControllerData.GetDigitalButtonState(xrc::Button::Menu);
+        const float leftTrigger = rawControllerData.GetLeftTriggerAdjustedForDeadZoneAndNormalized();
+        if (menuPressed)
+        {
+            if (!cheatWasPressed && leftTrigger > 0.9f)
+            {
+                cheatWasPressed = true;
+                OpenXRVk::xr_DebugDrawInput = !OpenXRVk::xr_DebugDrawInput;
+            }
+        }
+        else
+        {
+            cheatWasPressed = false;
+        }
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    void InputDeviceXRController::DrawGlobalDebugInfo()
+    {
+        CheckDebugDrawCheat();
+
+#if !defined(_RELEASE)
+        if (OpenXRVk::xr_DebugDrawInput)
+        {
+            DebugDisplayRequestBus::BusPtr debugDisplayBus;
+            DebugDisplayRequestBus::Bind(debugDisplayBus, g_defaultSceneEntityDebugDisplayId);
+            DebugDisplayRequests* debugDisplay{ DebugDisplayRequestBus::FindFirstHandler(debugDisplayBus) };
+            if (!debugDisplay || !IsSupported())
+            {
+                return;
+            }
+
+            // Save previous draw state
+            const AZ::u32 oldDrawState{ debugDisplay->GetState() };
+
+            // ... draw data to the screen ...
+            const auto& rawControllerData = m_impl->GetRawState();
+            DrawControllerAxes(*debugDisplay, rawControllerData.m_leftPositionState, rawControllerData.m_leftOrientationState);
+            DrawControllerAxes(*debugDisplay, rawControllerData.m_rightPositionState, rawControllerData.m_rightOrientationState);
+
+            float drawX = 20.f;     // current draw X
+            float drawY = 20.f;     // current draw Y
+            constexpr float textSize = 0.8f;
+            constexpr float lineHeight = 15.f;
+
+            const AZ::Color whiteColor{ 1.f, 1.f, 1.f, 1.f };
```

Comment:
[nit] There is already a white color defined in `AZ::Colors::White`

Reply:
Good to know, probably won't address it now.